firmware/library/L1_Drivers/button_driver.hpp

```diff
@@ -1,10 +1,87 @@
 #pragma once
 
-class ButtonInterface {
+#include <cstdint>
+
+#include "L0_LowLevel/LPC40xx.h"
+#include "L1_Drivers/gpio.hpp"
+#include "L1_Drivers/pin_configure.hpp"
+
+class ButtonInterface
+{
+ public:
+         virtual void Initialize()                        = 0;
+         virtual bool Read()                              = 0;
+         virtual bool Released()                          = 0;
+         virtual bool Pressed()                           = 0;
+         virtual void InvertButtonSignal(bool enable_invert_signal = true)  = 0;
+};
+
+class ButtonDriver : public ButtonInterface
```

Comment:
Would prefer the name `Button` rather than `ButtonDriver`. No other classes are named `Driver`.

Reply:
I think it would be logical to inherit the Gpio class rather than contain it.